src/libostree/ostree-repo-static-delta-compilation.c

```diff
@@ -608,7 +608,7 @@ try_content_rollsum (OstreeRepo *repo, DeltaOpts opts, const char *from, const c
                   (unsigned long long)matches->match_size);
     }
 
-  ContentRollsum *ret_rollsum = g_new0 (ContentRollsum, 1);
+  g_autofree ContentRollsum *ret_rollsum = g_new0 (ContentRollsum, 1);
   ret_rollsum->from_checksum = g_strdup (from);
   ret_rollsum->matches = g_steal_pointer (&matches);
   ot_transfer_out_value (out_rollsum, &ret_rollsum);
```

Comment:
That's not really going to work as `g_autofree` is only going to free the structure without the contents. If you really wanted to do this, you'd define a `g_autoptr` cleanup function pointing to `content_rollsums_free`. However, this almost certainly isn't needed since the allocation is transferred to the caller in `ot_transfer_out_value`.

It is good practice to declare a `g_autoptr` and then `NULL` the pointer before transferring it to the caller so that you can be sure that any early returns free the memory. However, just sticking in `g_autofree` isn't going to suffice here.